logs/logger.js

```diff
@@ -0,0 +1,30 @@
+// logger.js
+const winston = require("winston");
+
+// Define the log levels and colors (optional)
+const logLevels = {
+  error: "red",
+  warn: "yellow",
+  info: "green",
+  debug: "blue",
+};
+
+winston.addColors(logLevels);
+
+const logger = winston.createLogger({
+  level: "debug",
+  levels: logLevels,
+  format: winston.format.combine(
+    winston.format.colorize(),
+    winston.format.timestamp({ format: "YYYY-MM-DD HH:mm:ss" }),
+    winston.format.printf(({ timestamp, level, message }) => {
+      return `${timestamp} [${level.toUpperCase()}]: ${message}`;
```

Comment:
The `level.uppercase` is kinda showing up with weird blanks, because `colorize` is changing the value of `level` to include some extra text that tells the console how to color the word: winstonjs/winston#1345

what we can do if you want this to be capitalized is to set up the capitalization to happen earlier:

```javascript
const capitalizeLevel = winston.format((info) => {
  info.level = info.level.toUpperCase();
  return info;
});

winston.addColors(logLevels);

const logger = winston.createLogger({
  // level: "debug",
  levels: logLevels,
  format: winston.format.combine(
    capitalizeLevel(),
    winston.format.colorize(),
    winston.format.timestamp({ format: "YYYY-MM-DD HH:mm:ss" }),
    winston.format.printf(({ timestamp, level, message }) => {
      return `${timestamp} [${level}]: ${message}`;
    }),
  ),
  transports: [
    new winston.transports.Console(),
    new winston.transports.File({ filename: "logs/app.log" }),
  ],
});
```